hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -827,6 +827,18 @@ private boolean isKeyEmpty(OmKeyInfo keyInfo) {
         startKey, keyPrefix, maxKeys);
   }
 
+  @Override
+  public List<RepeatedOmKeyInfo> listTrash(String volumeName,
+      String bucketName, String startKeyName, String keyPrefix,
+      int maxKeys) throws IOException {
+
+    Preconditions.checkNotNull(volumeName);
+    Preconditions.checkNotNull(bucketName);
+
```

Comment:
As defined in Javadocs in KeyManager#listTrash:
`maxKeys - The number of keys to be returned. This must be below the cluster level set by admins.`
It would be great to add this check here.

Reply:
A preconditions check was added for the maxKeys to make sure that is below a cluster level max (new config).